samples/original-manual-adjusted.rb

```diff
@@ -30,7 +30,7 @@ def dewikify_hi(str, terms, intro = false)
       str = str
         .gsub(CODE_RE) { |x| code << x; "CODE#[#{code.length-1}]" }
         .gsub(/#{Regexp.quote(terms)}/i, '@\0@')
-        .gsub(/CODE#\[(\d+)\]/) { code[$1.to_i] }
+        .gsub(/CODE#\[(\d+)\]/) { code[Regexp.last_match(1).to_i] }
```

Comment:
you know what I think :D aka yes revise and explicitly use match or some other form that doesn't have to rely on global state